--- Image-Info-1.29/lib/Image/Info/JPEG.pm.orig	2009-07-08 22:18:22.000000000 +0200
+++ Image-Info-1.29/lib/Image/Info/JPEG.pm	2009-08-26 09:05:45.000000000 +0200
@@ -59,27 +59,6 @@
     $buf;
 }
 
-BEGIN {
-    my $f = ($] >= 5.008) ? <<'EOT' : <<'EOT';
-        sub with_io_string (&$) {
-	    open(my $fh, "<", \$_[1]);
-	    local $_ = $fh;
-	    &{$_[0]};
-	}
-EOT
-	sub with_io_string (&$) {
-	    require IO::String;
-	    local $_ = IO::String->new($_[1]);
-	    &{$_[0]};
-	    $_->close;
-	}
-EOT
-
-    #print $f;
-    eval $f;
-    die $@ if $@;
-}
-
 sub process_file
 {
     my($info, $fh, $cnf) = @_;
@@ -238,9 +217,9 @@
 
     if ($code == 0x10) {
 	eval {
-	    with_io_string {
-		_process_file($info, $_, 1);
-	    } $data;
+	    require IO::String;
+	    my $thumb_fh = IO::String->new($data);
+	    _process_file($info, $thumb_fh, 1);
 	};
 	$info->push_info(1, "error" => $@) if $@;
     }
@@ -283,9 +262,9 @@
 		my $jdata = substr($data, $ipos, $ilen);
 		#$info->push_info($i, "JPEGImage" => $jdata);
 
-		with_io_string {
-		    _process_file($info, $_, $i);
-		} $jdata;
+		require IO::String;
+		my $fh = IO::String->new($jdata);
+		_process_file($info, $fh, $i);
 	    }
 	}
 
